cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/AnalyzerUtils.java

```diff
@@ -48,21 +48,26 @@ private AnalyzerUtils() {
    * @param initialReplicaDistribution Initial distribution of replicas over the cluster.
    * @param initialLeaderDistribution Initial distribution of the leaders.
    * @param optimizedClusterModel The optimized cluster model.
+   * @param skipReplicationFactorChangeCheck Whether skip sanity check of topic partition's replication factor change before
+   *                                         and after optimization.
    * @return The diff represented by the set of balancing proposals to move from initial to final distribution.
    */
   public static Set<ExecutionProposal> getDiff(Map<TopicPartition, List<Integer>> initialReplicaDistribution,
                                                Map<TopicPartition, Integer> initialLeaderDistribution,
-                                               ClusterModel optimizedClusterModel) {
+                                               ClusterModel optimizedClusterModel,
+                                               boolean skipReplicationFactorChangeCheck) {
```

Comment:
Nit: It seems `skipReplicationFactorChangeCheck` is irrelevant to the majority of the existing users of this function. Can we keep the old one and create a new method with this extra parameter to be used by only the relevant callers?